src/ucp/api/ucp.h

```diff
@@ -1313,6 +1314,14 @@ typedef struct {
      * @ref ucp_atomic_op_nbx.
      */
     void          *reply_buffer;
+
+    /**
+     * Memory type of the buffer. see @ref ucs_memory_type_t for possible memory types.
+     * If this value is not set (along with its corresponding bit in the op_attr_mask -
+     * @ref UCP_OP_ATTR_FIELD_MEMORY_TYPE), then use default @ref UCS_MEMORY_TYPE_UNKNOWN
+     * which means the memory type will be detected automatically.
```

Comment:
If by default it auto-detects the memory type, why API user has to set anything ? 
Please add some description why it is advised to specify memory type ?

Reply:
It can avoid the cost of detection inside UCX if upper layer already knows the memory type,

Reply:
I know this :) I want to include just in the documentation so users understand why it is always preferable to have this specified.

Reply:
```
    /**
     * Memory type of the buffer. see @ref ucs_memory_type_t for possible memory types.
     * An optimization hint to avoid memory type detection for request buffer.
     * If this value is not set (along with its corresponding bit in the op_attr_mask -
     * @ref UCP_OP_ATTR_FIELD_MEMORY_TYPE), then use default @ref UCS_MEMORY_TYPE_UNKNOWN
     * which means the memory type will be detected internally.
     */

```

is this looks ok?

Reply:
Looks good